initrd/bin/kexec-seal-key

```diff
@@ -82,73 +84,131 @@ done
+			-ix 5 $pcr_5 \
+			-ix 6 $luks_pcr \
+			-ix 7 X
+	else
+		tpm sealfile2 \
+			-if "$KEY_FILE" \
+			-of "$TPM_SEALED" \
+			-hk 40000000 \
+			-ix 0 X \
+			-ix 1 X \
+			-ix 2 X \
+			-ix 3 X \
+			-ix 4 0000000000000000000000000000000000000000 \
+			-ix 5 $pcr_5 \
+			-ix 6 $luks_pcr \
+			-ix 7 X
+	fi
 
-	tpm nv_definespace \
-		-in $TPM_INDEX \
-		-sz $TPM_SIZE \
-		-pwdo "$tpm_password" \
-		-per 0 \
-	|| warn "Warning: Unable to define NVRAM space; trying anyway"
+	if [ $? -ne 0 ]; then
+		die "Unable to seal secret"
+	fi
 
+	shred -n 10 -z -u "$KEY_FILE" 2> /dev/null \
+		|| die "Failed to delete key file"
 
-	tpm nv_writevalue \
+	# try it without the owner password first
+	if ! tpm nv_writevalue \
 		-in $TPM_INDEX \
 		-if "$TPM_SEALED" \
-	|| die "Unable to write sealed secret to NVRAM"
+		; then
+		# to create an nvram space we need the TPM owner password
+		# and the TPM physical presence must be asserted.
+		#
+		# The permissions are 0 since there is nothing special
+		# about the sealed file
+		tpm physicalpresence -s \
+			|| warn "Warning: Unable to assert physical presence"
+
+		read -s -p "TPM Owner password: " tpm_password
+		echo
+
+		tpm nv_definespace \
+			-in $TPM_INDEX \
+			-sz $TPM_SIZE \
+			-pwdo "$tpm_password" \
+			-per 0 \
+			|| warn "Warning: Unable to define NVRAM space; trying anyway"
+
+
+		tpm nv_writevalue \
+			-in $TPM_INDEX \
+			-if "$TPM_SEALED" \
+			|| die "Unable to write sealed secret to NVRAM"
+	fi
+elif [ "$CONFIG_TPM2_TOOLS" = "y" ]; then
+	pcrf="/tmp/secret/pcrf.bin"
+	#TODO: verify logic below against tpm2
+	tpm2 pcrread -o "$pcrf" sha256:0,1,2,3
+	# pcr 4 is expected to be zero
+	dd if=/dev/zero bs=32 count=1 >> "$pcrf"
+	if [ "$CONFIG_USB_KEYBOARD" = "y" -o -r /lib/modules/libata.ko -o -x /bin/hotp_verification ]; then
+		# binary pcr dump can only go via stderr for redirection
+		# sadly this busybox has buggy support for "2>>" and "2|"
+		# so we need workaround...
+		tpm2 pcrread -o /dev/stderr sha256:5 2>&1 >/dev/console | cat >> "$pcrf"
+	else
+		dd if=/dev/zero bs=32 count=1 >> "$pcrf"
+	fi
+	# Use pcr 23 to precompute the value for pcr 6
+	tpmr extend -ix 23 -if /tmp/luksDump.txt
+	tpm2 pcrread -o /dev/stderr sha256:23 2>&1 >/dev/console | cat >> "$pcrf"
+	tpm2 pcrreset 23
+	tpm2 pcrread -o /dev/stderr sha256:7 2>&1 >/dev/console | cat >> "$pcrf"
+	if [ "$CONFIG_AUTO_UNLOCK" = "y" ]; then
+		#TODO: There is no CONFIG_AUTO_UNLOCK. Should be removed
+		tpmr seal "$KEY_FILE" "0x8100000$TPM_INDEX" sha256:0,1,2,3,4,5,6,7 "$pcrf"
+	else
+		#TODO: wrap TPM disk encryption key passphrase, otherwise prompt to user looks like if we were asking for TPM ownership passphrase
+		#TODO: everything is supposed to be under $pcrf, why considering them twice?
+		# TODO: review syntax to not duplicate expending pcr 2 times with pcr0-7: find a way to only use $pcrf? : sha256 "$pcrf" "$key_password"
+		tpmr seal "$KEY_FILE" "0x8100000$TPM_INDEX" sha256:0,1,2,3,4,5,6,7 "$pcrf" "$key_password"
```

Comment:
This is problematic and doesn work. Seems like we should use $pcrf but not PCRs0-7 here.
Doesn't unseal properly.